rule-engine/rule-engine-components/src/main/java/org/thingsboard/rule/engine/action/TbCreateAlarmNode.java

```diff
@@ -85,49 +79,45 @@ protected ListenableFuture<TbAlarmResult> processAlarm(TbContext ctx, TbMsg msg)
         final Alarm msgAlarm;
 
         if (!config.isUseMessageAlarmData()) {
-            alarmType = TbNodeUtils.processPattern(this.config.getAlarmType(), msg);
+            alarmType = TbNodeUtils.processPattern(config.getAlarmType(), msg);
             msgAlarm = null;
         } else {
             try {
-                msgAlarm = getAlarmFromMessage(ctx, msg);
+                msgAlarm = getAlarmFromMessage(ctx.getTenantId(), msg);
                 alarmType = msgAlarm.getType();
-            } catch (IOException e) {
-                ctx.tellFailure(msg, e);
-                return null;
+            } catch (IllegalArgumentException e) {
+                return Futures.immediateFailedFuture(e);
             }
         }
 
-        Alarm existingAlarm = ctx.getAlarmService().findLatestActiveByOriginatorAndType(ctx.getTenantId(), msg.getOriginator(), alarmType);
-        if (existingAlarm == null || existingAlarm.getStatus().isCleared()) {
-            return createNewAlarm(ctx, msg, msgAlarm);
-        } else {
-            return updateAlarm(ctx, msg, existingAlarm, msgAlarm);
-        }
+        ListenableFuture<Alarm> existingAlarmFuture = ctx.getDbCallbackExecutor().submit(
```

Comment:
Don't use DbCallbackExecutor to submit tasks. It is only for callbacks. Create a separate Async method in dao that uses jpaExecutor

Reply:
b46cc1e